File: LLD/DesignPatterns/Criteria/src/models/Product.java
package models;

public class Product {

    private String name;
    private Brand brand;
    private Category category;
    private double price;

    public Product(String name, double price, Brand brand, Category category) {
        this.name = name;
        this.price = price;
        this.brand = brand;
        this.category = category;
    }

    public String getName() {
        return name;
    }


    public Brand getBrand() {
        return brand;
    }


    public Category getCategory() {
        return category;
    }


    public double getPrice() {
        return price;
    }

}

File: LLD/InterviewExamples/Blinkit/Node/src/models/product.js
const { randomUUID } = require('crypto')

class Product {

    #id;
    #name;
    #category;
    #price;

    constructor(name,category,price) {
        this.#id = randomUUID()
        this.#name = name;
        this.#category = category;
        this.#price = price;
    }


    getName() {
        return this.#name;
    }

    getId() {
        return this.#id;
    }

    getCategory() {
        return this.#category;
    }

    getPrice() {
        return this.#price;
    }
}

module.exports = Product;